src/components/application_manager/include/application_manager/application_manager_impl.h

```diff
@@ -1705,6 +1718,9 @@ class ApplicationManagerImpl
   ServiceStreamingStatusMap streaming_application_services_;
   sync_primitives::Lock streaming_services_lock_;
 
+  mutable sync_primitives::Lock expired_button_requests_lock_;
+  mutable ExpiredButtonRequestsMap expired_button_requests_;
```

Comment:
Why does Core track expired requests instead of pending requests?

Reply:
@iCollin this is a tricky thing to align with the proposal statement:
> If after SDL Core sends a GENERIC_ERROR response to the application, HMI sends the response to Core notifying that the button has been successfully subscribed, then SDL Core shall send a request to HMI to unsubscribe the button. This is done to ensure the application and HMI are in sync.

So if the request is not expired, we can track HMI events in `on_event()` method of the corresponding request. However, if 
the request is expired, this object does not exist anymore in RequestController and if we still need to track the events from HMI after the corresponding request "death", then we need to keep their information in a separate container like that.

Reply:
Thank you, I see now, SubscribeButton response does not include the button info.